### PR TITLE
Improvments to metrics and hc

### PR DESCRIPTION
**Pull request by freesig**

### CODE REVIEW

**Comment by pdaoust** [comment]
hey @freesig I'm sorry I didn't look at the previous PR, and I don't know what sorts of pulls there are on this `hc` tool, but I thought I'd add a happ dev pull. I'm thinking about `hc generate` for creating a conductor config. Given that this is a replacement for an existing tool, and `hc generate` already has mindshare as the zome scaffolding tool, I'm wondering if this should have a different name -- like `hc config` or something. Is it too late for that?

File: crates/hc/src/bin/hc.rs [hunk 1]
```diff
@@ -103,35 +103,37 @@ async fn main() -> anyhow::Result<()> {
                 hc::force_admin_port(path, port)?;
             }
             if let Some(ports) = run {
-                run_n(&ops.holochain_path, paths, ports, ops.force_admin_ports).await?;
+                let holochain_path = ops.holochain_path.clone();
```
Comment (freesig) [inline crates/hc/src/bin/hc.rs:106]: keep

File: crates/hc/src/bin/hc.rs [hunk 2]
```diff
@@ -103,35 +103,37 @@ async fn main() -> anyhow::Result<()> {
...
+                hc::save::release_ports(std::env::current_dir()?).await?;
             }
         }
         Op::Run(Run { ports, existing }) => {
             let paths = existing.load()?;
             if paths.is_empty() {
                 return Ok(());
             }
-            run_n(&ops.holochain_path, paths, ports, ops.force_admin_ports).await?;
+            let holochain_path = ops.holochain_path.clone();
```
Comment (freesig) [inline crates/hc/src/bin/hc.rs:122]: keep

**Comment by freesig** [comment]
> hey @freesig I'm sorry I didn't look at the previous PR, and I don't know what sorts of pulls there are on this `hc` tool, but I thought I'd add a happ dev pull. I'm thinking about `hc generate` for creating a conductor config. Given that this is a replacement for an existing tool, and `hc generate` already has mindshare as the zome scaffolding tool, I'm wondering if this should have a different name -- like `hc config` or something. Is it too late for that?

I think @maackle is working on some name changes so maybe we can continue this convo in his hc PR

**Comment by maackle** [comment]
Yes @freesig @pdaoust #537 makes the command you're talking about `hc sandbox generate`. So there's still room for something like a `hc (dna|zome|app) generate`

**Pull request closed without merging**